### DISCUSSION TITLE
Dependent Fields in Laravel Nova not working with Repeater Fields

### DISCUSSION THREAD

**Discussion by DavidPuzder** [laravel/nova-issues#5669] [Ideas & Feature Requests] [answered]
- Laravel Version: v10.14.1
- Nova Version: v4.26.3
- PHP Version: 8.2
- Database Driver & Version: MariaDB 10.5
- Operating System and Version: OSX 12.6.3
- Browser type and version: Google Chrome (idk version)
- Reproduction Repository: (private)

Description:

The Dependent Fields functionality in Laravel Nova is not working properly when used in conjunction with the recently added Repeater Fields.

**Code Snippet:**
Inside the "Repeatable Field" section, the following code is used:

```php
/**
     * Get the fields displayed by the repeatable.
     *
     * @param  \Laravel\Nova\Http\Requests\NovaRequest  $request
     * @return array
     */
    public function fields(NovaRequest $request)
    {
        return [
            Select::make(__('attribute.singular'), 'attribute')
                ->options(Attribute::all()->pluck('name', 'id')->toArray())
                ->displayUsingLabels()
                ->searchable(),

            Select::make(__('attribute_value.singular'), 'attribute_value')
                ->hide()
                ->rules('sometimes')
                ->dependsOn('attribute', function ($field, NovaRequest $request, $formData) {
                    $field->readonly();

                    $attributeId = (string) $formData->resource(\App\Nova\Attribute::uriKey(), $formData->attribute);

                    if ($attributeId) {
                        $values = \App\Models\AttributeValue::where(['attribute_id' => $attributeId])->pluck('value', 'id')->toArray();
                        $field->options($values)->rules('required')->show()->readonly(false);
                        return;
                    }
                })
                ->displayUsingLabels()
                ->searchable(),
        ];
    }
```

**Issue Details:**
The code mentioned above works perfectly when used in the `fields` function inside a regular `Resource` class. However, when used inside the Repeatable Field, the second select field does not appear, and even if it is initially shown, it will hide after manipulating the first select field.

**Steps to Reproduce:**
1. Install a fresh Laravel Nova installation.
2. Use the provided code snippet inside a Repeatable Field.
3. Attempt to use the Dependent Fields functionality with the Repeater Fields.
4. Notice that the second select field does not show up and disappears after manipulating the first select field.

Please let me know if you need any further information or clarification.

**Comment by DavidPuzder** [comment]
Code insile `Resource` class:(this one works)

```php
Repeater::make(__('attribute_value.plural'), 'attribute_values', AttributeValue::class)
     ->repeatables([
          AttributeValueItem::make(),
     ])
     ->asJson(),
```

**Reply by niunaiyi** [reply]
it work? AttributeValueItem can DependsOn?

**Comment by kichetof** [comment, marked as answer]
Unfortunately, it's not (yet?) supported:
`Also, fields inside a Repeatable do not support dependent field (dependsOn) functionality.`


https://nova.laravel.com/docs/4.0/resources/repeater-fields.html#repeater-field-capabilities

**Comment by danish-khan-I** [comment]
Is there an ETA for dependsOn with Repeatable fields? or any workaround.

**Reply by nikolai-vysotskyi** [reply]
Is there any workaround?

**Comment by metadeck** [comment]
`Also, fields inside a Repeatable do not support dependent field (dependsOn) functionality.`

Could we have the repeatable field itself to be compatible with dependsOn as they are on the same form and same level?

For context I'm building a form builder resource, and would like to show the repeatable field based on the value of a select on the same form.

```php
    public function fields(NovaRequest $request)
    {
        return [
            Select::make('Type')
                ->options(FormFieldType::dropdownOptions())
                ->displayUsingLabels()
                ->required(),
            Repeater::make('Options')
                ->dependsOn('type', FormFieldType::SELECT) // this function doesn't exist
                ->repeatables([
                    SelectOption::make(),
                ])
                ->asJson(),
        ];
    }
```

**Reply by RomainMazB** [reply]
I need this level of `dependsOn` too

**Comment by myounis97** [comment]
We really need this feature besides nested repeatable

**Comment by itrack** [comment]
+1 I think it is a very necessary functionality

**Comment by kennethtomagan** [comment]
+ 1 we need this functionality in repeater

**Comment by nadlambino** [comment]
+1

**Comment by metadeck** [comment]
Would this be on the roadmap? Even just the ability to hide the Repeater field itself based on the value of another field?

**Reply by nadlambino** [reply]
With the recent about the framework and company itself, I hope this will be in the roadmap + more features as well.

**Comment by lorenzoalulithos** [comment]
+ 1 :)

**Comment by jase-languasco** [comment]
+1

**Comment by maul75** [comment]
Is this also not planned for nova 5?

**Comment by haythoo** [comment]
Hi, any plan for this feature?

**Comment by andristet** [comment]
Waiting for this feature. At least for the Repeater itself, not the fields inside it.
